### PR TITLE
Add vlan_tag resource and data source (#48)

### PR DESCRIPTION
**Pull request by wras66**

### CODE REVIEW

File: gen/definitions/vlan_tag.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  - model_name: description
+    type: String
+    description: Optional user-created description.
+    example: "My TAG id"
+    mandatory: true
+  - model_name: overridable
+    type: Bool
+    description: Indicates whether object values can be overridden.
+    exclude_example: true
+    test_value: "true"
```
Comment (danischm) [inline gen/definitions/vlan_tag.yaml:21]: We normally just include an `example` and don't require a `test_value` to be defined. `test_value` is only required if we need to make use of value only available at runtime, ie. something that needs to be retrieved via a data source (in pre-requisites).

File: gen/definitions/vlan_tag.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    minimum_test_value: "true"
+  - model_name: type
+    type: String
+    mandatory: true
+    example: "VlanTag"
+  - model_name: startTag
```
Comment (danischm) [inline gen/definitions/vlan_tag.yaml:27]: I think this one should be `mandatory`.

File: gen/definitions/vlan_tag.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    type: String
+    example: "12"
+    test_value: "12"
+    data_path: ["data"]
+  - model_name: endTag
```
Comment (danischm) [inline gen/definitions/vlan_tag.yaml:32]: Not sure if this one should be mandatory as well.

File: gen/definitions/vlan_tag.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    type: String
+    description: Optional user-created description.
+    example: "My TAG id"
+    mandatory: true
+  - model_name: overridable
+    type: Bool
+    description: Indicates whether object values can be overridden.
+    exclude_example: true
+    test_value: "true"
+    minimum_test_value: "true"
```
Comment (danischm) [inline gen/definitions/vlan_tag.yaml:22]: This should not be required for the "minimal" test.

File: gen/definitions/vlan_tag.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  - model_name: type
```
Comment (danischm) [inline gen/definitions/vlan_tag.yaml:23]: Could you try defining this attribute like this:

```yaml
- model_name: type
  type: String
  value: "VlanTag"
```

This should hardcode the value and not expose it as a resource attribute.

File: gen/definitions/vlan_tag.yaml (hunk 1 above)
Comment (wras66) [reply gen/definitions/vlan_tag.yaml]: There are two scenarios for VLAN_TAG creation:
1. forcing to set override to TRUE and skipping startTag and endTag
2. setting startTag and endTag and living override
both method tested,
finally i have changed according to your request, that case is more useful and natural

File: gen/definitions/vlan_tag.yaml (hunk 2 above)
Comment (wras66) [reply gen/definitions/vlan_tag.yaml:24]: Done

File: gen/definitions/vlan_tag.yaml (hunk 3 above)
Comment (wras66) [reply gen/definitions/vlan_tag.yaml:30]: Done

File: gen/definitions/vlan_tag.yaml (hunk 5 above)
Comment (wras66) [reply gen/definitions/vlan_tag.yaml:21]: Done

**Review by danischm — APPROVED** [review]

**Pull request merged**